sign/sign.go

```diff
@@ -173,11 +174,23 @@ func (s *Signer) SignFile(path string) (*SignedObject, error) {
 		return nil, fmt.Errorf("sign file path: %s: %w", path, err)
 	}
 
-	object, err := s.impl.VerifyFileInternal(s, path)
-	if err != nil {
-		return nil, fmt.Errorf("verify file path: %s: %w", path, err)
+	// After signing, registry consistency may not be there right
+	// away. Retry the image verification if it fails
+	// ref: https://github.com/kubernetes-sigs/promo-tools/issues/536
+	i := 0
+	for {
+		object, err := s.impl.VerifyFileInternal(s, path)
+		if err != nil {
+			i++
+			if i <= s.options.MaxRetries {
+				s.options.Logger.Info("Signature could not be verified, retrying")
+				time.Sleep(time.Duration(i) * time.Second)
```

Comment:
Ultra cool, but maybe too much: we could use a backoff here like this:
https://github.com/kubernetes-sigs/security-profiles-operator/blob/de3017ad869867efb9743a1c387398e7741fd3cc/internal/pkg/util/retries.go#L48

Reply:
Sure, go ahead! :)

Reply:
Awesome, let me think of a way to abstract it and I'll add a new package. Meanwhile, I've added the expbackoff to the SignImage function. PTAL :pray: